### PR TITLE
fix: page response missing CSP and Link headers when return promise in `load`

### LINKED ISSUE #11801: Streaming disables CSP silently
### Describe the bug

Using streaming disables the "built in" CSP support and this is not documented clearly (reading both the CSP and streaming doc). The workaround is to use my own CSP in a handle hook, but to support streaming I have to add `script-src: 'unsafe-inline'` which is undesirable. I was not able to find the nonce in the handle hook that is referenced in the doc.

### Reproduction

https://github.com/vegardok/sveltekit-csp-and-streaming/commits/main/

### Logs

_No response_

### System Info

```Shell
System:
    OS: macOS 14.3
    CPU: (10) arm64 Apple M2 Pro
    Memory: 295.05 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.9.0 - ~/.nvm/versions/node/v20.9.0/bin/node
    npm: 10.1.0 - ~/.nvm/versions/node/v20.9.0/bin/npm
    pnpm: 8.14.0 - ~/.nvm/versions/node/v20.9.0/bin/pnpm
  Browsers:
    Chrome: 121.0.6167.85
    Safari: 17.3
  npmPackages:
    @sveltejs/adapter-auto: ^3.0.0 => 3.1.1 
    @sveltejs/kit: ^2.0.0 => 2.5.0 
    @sveltejs/vite-plugin-svelte: ^3.0.0 => 3.0.2 
    svelte: ^4.2.7 => 4.2.9 
    vite: ^5.0.3 => 5.0.12
```

### Severity

serious, but I can work around it

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by 0x221A**
closes #11801 

---

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [x] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. Changesets that add features should be `minor` and those that fix bugs should be `patch`. Please prefix changeset messages with `feat:`, `fix:`, or `chore:`.

### Edits

- [x] Please ensure that 'Allow edits from maintainers' is checked. PRs without this option may be closed.

### CODE REVIEW

**Comment by darg-2036** [comment]
Would love to see this merged

File: packages/kit/src/runtime/server/page/render.js [hunk 1]
```diff
@@ -511,9 +512,7 @@ export async function render_response({
 					type: 'bytes'
 				}),
 				{
-					headers: {
-						'content-type': 'text/html'
-					}
+					headers
```
Comment (eltigerchino) [inline packages/kit/src/runtime/server/page/render.js:515]: What's the reason for this change?

**Comment by eltigerchino** [comment]
Thank you for the PR. I think it's good to merge but is it possible if we add a test to check that the CSP headers are included in the server response?

**Review by eltigerchino — APPROVED** [review]

**Comment by 0x221A** [comment]
@eltigerchino sure!

**Comment by eltigerchino** [comment]
Thank you so much!

**Pull request merged**